ipaclient/plugins/dns.py

```diff
@@ -417,6 +418,63 @@ def interactive_prompt_callback(self, kw):
 
 @register(override=True, no_fail=True)
 class dns_update_system_records(MethodOverride):
+    record_groups = ('ipa_records', 'location_records')
+
+    def get_options(self):
+        for option in super(dns_update_system_records, self).get_options():
+            yield option
+        yield Str(
+            'out?',
+            include='cli',
+            doc=_('file to store DNS records in nsupdate format')
+        )
+
+    def _standard_output(self, textui, result, labels):
+        """Print output in standard format common across the other plugins"""
+        for key in self.record_groups:
+            if result.get(key):
+                textui.print_indented(u'{}:'.format(labels[key]), indent=1)
+                for val in sorted(result[key]):
+                    textui.print_indented(val, indent=2)
+                textui.print_line(u'')
+
+    def _nsupdate_output_file(self, file, result, labels):
+        """Store data in nsupdate format in file"""
+        def parse_rname_rtype(record):
+            """Get rname and rtype from textual representation of record"""
+            l = record.split(' ', 5)
+            return l[0], l[3]
+
+        already_removed = set()
+        for key in self.record_groups:
+            if result.get(key):
+                file.write("; {}\n".format(labels[key]))  # comment
+                for val in sorted(result[key]):
+                    # delete old first
+                    r_name_type = parse_rname_rtype(val)
+                    if r_name_type not in already_removed:
+                        # remove it only once
+                        already_removed.add(r_name_type)
+                        file.write("update delete {rname} {rtype}\n".format(
+                            rname=r_name_type[0], rtype=r_name_type[1]
+                        ))
+                    # add new
+                    file.write("update add {}\n".format(val))
```

Comment:
So the command `update delete ...` will remove for example **all** A records for a domain and that's why you're checking that it won't run twice. However, when the records are added, they should be all different (in case of A record, different IPv4). Is that correct?